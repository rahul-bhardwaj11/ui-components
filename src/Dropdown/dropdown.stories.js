import React from 'react';

import { storiesOf } from '@storybook/react';
import Dropdown from './index';
import { withInfo } from '@storybook/addon-info';
import { withKnobs, object } from '@storybook/addon-knobs';
import { Icon } from 'antd';

const stories = storiesOf('Dropdown', module);
stories.addDecorator(withKnobs);

const options = [
  { key: '1', content: 'First Item' },
  { key: '2', content: 'Second Item' },
  { key: '3', content: 'Third Item', cautious: true }
];

stories
  .add(
    'Default Dropdown',
    withInfo('Basic usage of the Dropdown')(() => (
      <Dropdown options={object('options', options)}>
        <a className="ant-dropdown-link" href="#">
          Hover me
        </a>
      </Dropdown>
    ))
  )
  .add(
    'Button Dropdown',
    withInfo('Basic usage of the Dropdown')(() => (
      <Dropdown
        options={object('options', options)}
        type="button"
        label={'button'}
      />
    ))
  )

  .add(
    'Link Dropdown',
    withInfo('Basic usage of the Dropdown')(() => (
      <Dropdown options={object('options', options)} trigger={'click'}>
        <a className="ant-dropdown-link" href="#">
          Click me <Icon type="down" />
        </a>
      </Dropdown>
    ))
  );
