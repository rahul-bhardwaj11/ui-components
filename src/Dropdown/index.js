import React, { Component } from 'react';
import PropTypes from 'prop-types';
import AntDropdown from 'antd/lib/dropdown';
import Menu from '../Menu';
import 'antd/lib/dropdown/style/index.css';
import Button from 'antd/lib/button';
import styled from 'styled-components';
import 'antd/lib/button/style/index.css';

const MtWrapper = styled.div`
  display: inline-block;
`;

class Dropdown extends Component {
  static propTypes = {
    options: PropTypes.any.isRequired,
    trigger: PropTypes.string,
    children: PropTypes.node,
    onSelect: PropTypes.func,
    type: PropTypes.string,
    label: PropTypes.string,
    getPopupContainer: PropTypes.func,
    placement: PropTypes.string,
    className: PropTypes.string,
    selectedKeys: PropTypes.array
  };

  static defaultProps = {
    trigger: 'hover',
    placement: 'bottomRight',
    onSelect: () => {}
  };
  dropdownRef = null;

  getPopupContainer = () => this.dropdownRef;

  dropdownRefCallback = e => e && (this.dropdownRef = e);

  render() {
    let {
      options,
      trigger,
      children,
      type,
      label,
      onSelect,
      placement,
      className,
      selectedKeys,
      getPopupContainer = this.getPopupContainer,
      ...dropdownProps
    } = this.props;

    let overlay;
    if (options instanceof Array) {
      overlay = (
        <Menu
          options={options}
          onClick={onSelect}
          selectedKeys={selectedKeys}
        />
      );
    } else {
      overlay = options;
    }
    if (type === 'button') {
      children = <Button>{label || 'Button'}</Button>;
    }

    return (
      <React.Fragment>
        <MtWrapper className={className} innerRef={this.dropdownRefCallback}>
          <AntDropdown
            overlay={overlay}
            trigger={[trigger]}
            prefixCls={'ant-dropdown'}
            getPopupContainer={getPopupContainer}
            placement={placement}
            {...dropdownProps}
          >
            {children}
          </AntDropdown>
        </MtWrapper>
      </React.Fragment>
    );
  }
}
export default Dropdown;
