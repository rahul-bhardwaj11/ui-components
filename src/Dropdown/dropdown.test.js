import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Dropdown from './index';

describe('Dropdown', () => {
  const options = [
    { key: '1', content: 'First Item' },
    { key: '2', content: 'Second Item' },
    { key: '3', content: 'Third Item' }
  ];
  it('matches default snapshot', () => {
    const tree = renderer.create(<Dropdown options={options} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches Button Dropdown snapshot', () => {
    const tree = renderer
      .create(<Dropdown options={options} type="button" label={'button'} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches Link Dropdown snapshot', () => {
    const tree = renderer
      .create(<Dropdown options={options} trigger={'click'} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should handle click', () => {
    const clickSpy = jest.fn();
    const dropDown = shallow(<Dropdown options={options} onClick={clickSpy} />);
    dropDown.find('Dropdown').simulate('click');
    expect(clickSpy).toHaveBeenCalled();
  });
});
