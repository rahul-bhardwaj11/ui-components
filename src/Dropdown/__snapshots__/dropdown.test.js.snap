// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown matches Button Dropdown snapshot 1`] = `
<div
  className="ant-dropdown-trigger"
  disabled={undefined}
  onBlur={undefined}
  onClick={undefined}
  onContextMenu={undefined}
  onFocus={undefined}
  onMouseDown={undefined}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={undefined}
>
   
  <button
    className="ant-btn"
    onClick={[Function]}
    type="button"
  >
    <span>
      button
    </span>
  </button>
   
</div>
`;

exports[`Dropdown matches Link Dropdown snapshot 1`] = `
<div
  className="ant-dropdown-trigger"
  disabled={undefined}
  onBlur={undefined}
  onClick={[Function]}
  onContextMenu={undefined}
  onFocus={undefined}
  onMouseDown={[Function]}
  onMouseEnter={undefined}
  onMouseLeave={undefined}
  onTouchStart={[Function]}
>
   
   
</div>
`;

exports[`Dropdown matches default snapshot 1`] = `
<div
  className="ant-dropdown-trigger"
  disabled={undefined}
  onBlur={undefined}
  onClick={undefined}
  onContextMenu={undefined}
  onFocus={undefined}
  onMouseDown={undefined}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onTouchStart={undefined}
>
   
   
</div>
`;
